### ISSUE TITLE
Special-case postmaster address

### ISSUE THREAD

**Issue by foxcpp** [foxcpp/maddy#80]
```
RCPT TO:<postmaster>
```
Should be accepted as valid.
There are several places around the code that assume user@domain in recipients addresses, need to check them and probably move some code around.

See https://tools.ietf.org/html/rfc5321#page-14 and https://tools.ietf.org/html/rfc5321#section-4.5.1.

**Comment by foxcpp** [comment]
- [ ] Adjust user@domain handling all around the code to accept "postmaster" as a valid address
- [ ] Make destination pipeline directive/step understand <postmaster> address as a special case
- [ ] postmaster address should be case-insensitive, see #78

**Comment by foxcpp** [comment]
Or better, it should be implemented using `rewrite` directive defined in #18.
```
rewrite postmaster postmaster@example.org
```
